sunpy/database/database.py

```diff
@@ -226,7 +227,17 @@ def __init__(self, url=None, CacheClass=LRUCache, cache_size=float('inf'),
         self._session_cls = sessionmaker(bind=self._engine)
         self.session = self._session_cls()
         self._command_manager = commands.CommandManager()
-        self.default_waveunit = default_waveunit
+
+        if default_waveunit:
+            if isinstance(default_waveunit, units.Unit):
+                self.default_waveunit = default_waveunit    
+            elif isinstance(default_waveunit, str):
```

Comment:
we should drop support for string here, you can just throw everything through Unit()